otx/api/entities/label_schema.py

```diff
@@ -558,6 +558,68 @@ def from_labels(cls, labels: Sequence[LabelEntity]) -> "LabelSchemaEntity":
         label_group = LabelGroup(name="from_label_list", labels=labels)
         return LabelSchemaEntity(label_groups=[label_group])
 
+    def resolve_labels_greedily(self, scored_labels: List[ScoredLabel]) -> List[ScoredLabel]:
+        """Resolves hierarchical labels and exclusivity based on a list of ScoredLabels (labels with probability).
+
+        The following two steps are taken:
+
+        - select the most likely label from each label group
+        - add their predecessors if they are also most likely labels (greedy approach).
```

Comment:
I would also add: "skip the candidate label if it is a successor of the label from another group that has a low probability"

Reply:
I'd say that this statement follows from the previous one, but my last formulation was a bit unclear.